### PR TITLE
Added Password Strength Validation (8 Characters, Special Character, and Digit)

### PR DESCRIPTION
**Pull request by jaidh01**
Closes:  #118 

- **Title**:  Added Password Strength Validation (8 Characters, Special Character, and Digit)
- **Your Name**: Jai Dhingra
- **Idenitfy yourself**: GSSOC-extd Contributor, Hacktoberfest

### Describe the add-ons or changes you've made 📃

Added Password strength Validation (8 Characters, Special Character, and Digit).

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/Jarvis/blob/main/Contributing.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/Jarvis/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷

https://github.com/user-attachments/assets/653896a8-06da-43f1-8c31-3703b8a94ab6

### CODE REVIEW

File: src/auth/signup.py
Comment (Avdhesh-Varshney) [inline src/auth/signup.py:1]: Keep the file in 2 space seprated code and commit your code only. It'll be difficult to review the pr

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]

File: src/auth/signup.py
Comment (jaidh01) [reply src/auth/signup.py:1]: Sorry for the inconvenience.
I have done all these changes. Please spare a time to look at it.

**Comment by jaidh01** [comment]
Hi @Avdhesh-Varshney, I have done all the changes as you suggested.
Please spare a time to look at it.

**Comment by jaidh01** [comment]
Hi @Avdhesh-Varshney, Is it good, or some changes should be done?

**Comment by jaidh01** [comment]
Hi @Avdhesh-Varshney @Zedoman, Is there anything else I have to do in this?

**Review by Zedoman — APPROVED** [review]

**Review by Avdhesh-Varshney — APPROVED** [review]

**Pull request merged**